### PR TITLE
Bug 1707323: Operator may attempt to write into nil map

### PR DESCRIPTION
**Pull request by josefkarasek**
```
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```

### CODE REVIEW

**Comment by pweil-** [comment]
> add tests
Awesome, thanks! :smile:

File: pkg/k8shandler/util.go [hunk 1]
```diff
@@ -149,7 +159,7 @@ func isValidConf(dpl *api.Elasticsearch) error {
 		if err := updateConditionWithRetry(dpl, v1.ConditionTrue, updateInvalidReplicationCondition); err != nil {
 			return err
 		}
-		return fmt.Errorf("Wrong RedundancyPolicy selected. Choose different RedundancyPolicy or add more nodes with data roles")
+		return fmt.Errorf("Wrong RedundancyPolicy selected '%s'. Choose different RedundancyPolicy or add more nodes with data roles", dpl.Spec.RedundancyPolicy)
```
Comment (ewolinetz) [inline pkg/k8shandler/util.go]: can we make this a condition update instead?

File: pkg/k8shandler/util_test.go [hunk 2]
```diff
@@ -104,3 +106,98 @@ func TestSelectorsCommonOverwritten(t *testing.T) {
...
+func TestNoRedundancyPolicySpecified(t *testing.T) {
+
+	esCR := &api.Elasticsearch{
+		Spec: api.ElasticsearchSpec{},
+	}
+
+	isValid := isValidRedundancyPolicy(esCR)
+
+	if !isValid {
+		t.Errorf("Users don't have to select redundancy policy. SingleRedundancy is the default")
```
Comment (ewolinetz) [inline pkg/k8shandler/util_test.go]: "Expected default policy of SingleRedundancy to be used, incorrectly flagged as invalid"

File: pkg/k8shandler/util_test.go [hunk 3]
```diff
@@ -104,3 +106,98 @@ func TestSelectorsCommonOverwritten(t *testing.T) {
...
+	esCR := &api.Elasticsearch{
+		Spec: api.ElasticsearchSpec{
+			RedundancyPolicy: api.FullRedundancy,
+		},
+	}
+
+	isValid := isValidRedundancyPolicy(esCR)
+
+	if !isValid {
+		t.Error("Valid policy flagged as invalid")
```
Comment (ewolinetz) [inline pkg/k8shandler/util_test.go]: "Expected FullRedundancy to be valid, flagged as invalid"

File: pkg/k8shandler/util_test.go [hunk 4]
```diff
@@ -104,3 +106,98 @@ func TestSelectorsCommonOverwritten(t *testing.T) {
...
+	esCR := &api.Elasticsearch{
+		Spec: api.ElasticsearchSpec{
+			RedundancyPolicy: "NoRedundancy",
+		},
+	}
+
+	isValid := isValidRedundancyPolicy(esCR)
+
+	if isValid {
+		t.Error("Invalid policy wasn't caught")
```
Comment (ewolinetz) [inline pkg/k8shandler/util_test.go]: Expected NoRedundancy to be flagged as invalid, was found to be valid.

File: pkg/k8shandler/util_test.go [hunk 5]
```diff
@@ -104,3 +106,98 @@ func TestSelectorsCommonOverwritten(t *testing.T) {
...
+		Spec: api.ElasticsearchSpec{
+			Nodes: []api.ElasticsearchNode{esNode},
+		},
+	}
+
+	rc := calculateReplicaCount(esCR)
+
+	// we default to 1
+	if rc != 1 {
+		t.Errorf("Users don't have to select redundancy policy. SingleRedundancy is the default")
```
Comment (ewolinetz) [inline pkg/k8shandler/util_test.go]: Please rephrase this to match other unit test error messages.

**Comment by ewolinetz** [comment]
overall looks good, requesting some rephrasing of the test failures to match other existing tets.
